src/Simplify.cpp

```diff
@@ -4274,6 +4349,16 @@ void check_algebra() {
     check((x - y)*-2, (y - x)*2);
     check((xf - yf)*-2.0f, (yf - xf)*2.0f);
 
+    // Pull terms that are a multiple of the divisor out of a ternary expression
+    check(((x*4 + y) + z) / 2, x*2 + (y + z)/2);
+    check(((x*4 - y) + z) / 2, x*2 - (y - z)/2);
```

Comment:
(-a)/2 is not the same as -(a/2), so I think this rule is wrong:

If x ==0, y == 2, and z == 1, the LHS is 

((0*4 - 2) + 1) / 2 == (-1)/2 == -1

and the RHS is

0*2 - (2 - 1)/2 == -(1/2) == 0

(integer division always rounds down in Halide for positive denominators)